### DISCUSSION TITLE
HTTPX 0.27.0 adds a deprecation message when using TestClient

### DISCUSSION THREAD

**Discussion by Gcraft25** [fastapi/fastapi#11183] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.testclient import TestClient

def test_deprecation_message():
    app = FastAPI()
    client = TestClient(app)
    return client
```

### Description

With the release of HTTPX 0.27.0 on 2024-02-21, there is now a deprecation warning when creating the TestClient. The message is as follows

`DeprecationWarning: The 'app' shortcut is now deprecated. Use the explicit style 'transport=WSGITransport(app=...)' instead.`

Running the above code using Pytest reproduces the issue. 

### Operating System

macOS

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

3.11.3

**Comment by jgould22** [comment, marked as answer]
The TestClient comes from Starlette.

You can see the issue here https://github.com/encode/starlette/issues/2524

**Reply by Gcraft25** [reply]
Thank you!

**Reply by dodumosu** [reply]
as at starlette 0.37.0, this is fixed, but since the version of starlette allowed in the dependencies specifically excludes 0.37.0, the issue will persist: encode/starlette#2523
